providers/implementations/keymgmt/dsa_kmgmt.c

```diff
@@ -199,6 +199,9 @@ static int dsa_export(void *keydata, int selection, OSSL_CALLBACK *param_cb,
     if (!ossl_prov_is_running() || dsa == NULL)
         goto err;
 
+    if (tmpl == NULL)
+        return ok;
```

Comment:
This is wrong.  And apparently the goto err above is wrong as well, because ok should be 0 initially and only later set to 1.

Reply:
So I later change the initial value of ok to 0?

Reply:
I would initialize it to 0 and only after this if I'd change it to 1.

Reply:
Or, what you did now. That's even better.